testnet/stacks-node/src/run_loop/nakamoto.rs

```diff
@@ -0,0 +1,1023 @@
+                        .require_affirmed_anchor_blocks,
+                    ..ChainsCoordinatorConfig::new()
+                };
+                ChainsCoordinator::run(
+                    coord_config,
+                    chain_state_db,
+                    moved_burnchain_config,
+                    &mut coordinator_dispatcher,
+                    coordinator_receivers,
+                    moved_atlas_config,
+                    cost_estimator.as_deref_mut(),
+                    fee_estimator.as_deref_mut(),
+                    miner_status,
+                    coordinator_indexer,
+                    atlas_db,
+                );
+            })
+            .expect("FATAL: failed to start chains coordinator thread");
+
+        coordinator_thread_handle
+    }
+
+    /// Start Prometheus logging
+    fn start_prometheus(&mut self) {
+        let prometheus_bind = self.config.node.prometheus_bind.clone();
+        if let Some(prometheus_bind) = prometheus_bind {
+            thread::Builder::new()
+                .name("prometheus".to_string())
+                .spawn(move || {
+                    debug!("prometheus thread ID is {:?}", thread::current().id());
+                    start_serving_monitoring_metrics(prometheus_bind);
+                })
+                .unwrap();
+        }
+    }
+
+    /// Get the sortition DB's highest block height, aligned to a reward cycle boundary, and the
+    /// highest sortition.
+    /// Returns (height at rc start, sortition)
+    fn get_reward_cycle_sortition_db_height(
+        sortdb: &SortitionDB,
+        burnchain_config: &Burnchain,
+    ) -> (u64, BlockSnapshot) {
+        let (stacks_ch, _) = SortitionDB::get_canonical_stacks_chain_tip_hash(sortdb.conn())
+            .expect("BUG: failed to load canonical stacks chain tip hash");
+
+        let sn = match SortitionDB::get_block_snapshot_consensus(sortdb.conn(), &stacks_ch)
+            .expect("BUG: failed to query sortition DB")
+        {
+            Some(sn) => sn,
+            None => {
+                debug!("No canonical stacks chain tip hash present");
+                let sn = SortitionDB::get_first_block_snapshot(&sortdb.conn())
+                    .expect("BUG: failed to get first-ever block snapshot");
+                sn
+            }
+        };
+
+        (
+            burnchain_config.reward_cycle_to_block_height(
+                burnchain_config
+                    .block_height_to_reward_cycle(sn.block_height)
+                    .expect("BUG: snapshot preceeds first reward cycle"),
+            ),
+            sn,
+        )
+    }
+
+    /// Wake up and drive stacks block processing if there's been a PoX reorg.
+    /// Be careful not to saturate calls to announce new stacks blocks, because that will disable
+    /// mining (which would prevent a miner attempting to fix a hidden PoX anchor block from making
+    /// progress).
+    fn drive_pox_reorg_stacks_block_processing(
```

Comment:
This can be vastly simplified now that affirmation maps don't omit anchor blocks.

Reply:
I removed the chain liveness thread (and its functions) for now.